feat(Insomnia): SCIM auto refresh updates (#5363)
Kong/developer.konghq.com: app/insomnia/scim.md

---

> title: "System for Cross-domain Identity Management (SCIM) for {{ site.data.products.insomnia.name }}"

Reviewer: This variable works, but ideally we'd specify the variable in this file: jekyll.yml and use `{{site.insomnia}}`. This is out of scope for this PR though.

Author: Created issue to change that: https://github.com/Kong/developer.konghq.com/issues/5545

    - insomnia
tier: enterprise
related_resources:
  - text: "SSO for {{ site.data.products.insomnia.name }}"
    url: /insomnia/sso/
  - text: "{{ site.data.products.insomnia.name }} Enterprise"
    url: /insomnia/enterprise/
  - text: Enterprise user management
    url: /insomnia/enterprise-user-management/
  - text: Enterprise account management
    url: /insomnia/enterprise-account-management/
  - text: "Configure SCIM for {{ site.data.products.insomnia.name }} with Okta"
    url: /how-to/configure-scim-for-insomnia-with-okta/
  - text: "Configure SCIM for {{ site.data.products.insomnia.name }} with Azure"
    url: /how-to/configure-scim-for-insomnia-with-azure/    

faqs:
  - q: Do SCIM tokens expire?
    a: |
      Yes. SCIM tokens can expire depending on the refresh method you chose when configuring SCIM:
      
      - Manual refresh: the token expires at the expiration date you pick.

>       - Automatic refresh (requires OAuth 2.0): the token expires and {{ site.data.products.insomnia.name }} automatically refreshes it.

Reviewer: ```suggestion
      - Automatic refresh (requires OAuth 2.0) {% new_in 13 %}: the token expires and {{ site.data.products.insomnia.name }} automatically refreshes it.
```
This badge should work, you might want to double check the output.

When you unassign, deactivate, or delete a user in your IdP, the IdP sends a provisioning update to {{ site.data.products.insomnia.name }}.

When SCIM is enabled, the following users consume {{ site.data.products.insomnia.name }} licenses:
- Users provisioned through SCIM consume Enterprise licenses.
- Manually added users continue to consume licenses until you remove them or transition them to IdP-managed provisioning.

For more information about license management, see [Enterprise user management](/insomnia/enterprise-user-management/).

## SCIM connector token lifecycle
SCIM provisioning uses a connector URL and token generated in {{ site.data.products.insomnia.name }}. The token authorizes your identity provider to provision users and teams.

Administrators can view the current SCIM token status in {{ site.data.products.insomnia.name }}:
1. From the {{ site.data.products.insomnia.name }} Enterprise control dashboard sidebar, click [**SCIM**](https://app.insomnia.rest/app/enterprise/scim).
1. Review the SCIM configuration page to see:
  - If SCIM is enabled.
  - If the token is valid, expiring soon, or expired.

> Configuring SCIM with OAuth 2.0 lets {{ site.data.products.insomnia.name }} automatically refresh the token. When a token is close to expiration and cannot be refreshed automatically, {{ site.data.products.insomnia.name }} displays a warning message on the SCIM page and sends email notifications starting 20 days before the token expires.

Reviewer: ```suggestion
{% new_in 13 %} Configuring SCIM with OAuth 2.0 lets {{ site.data.products.insomnia.name }} automatically refresh the token. When a token is close to expiration and cannot be refreshed automatically, {{ site.data.products.insomnia.name }} displays a warning message on the SCIM page and sends email notifications starting 20 days before the token expires.
```

> {% new_in 13 %} Configuring SCIM with OAuth 2.0 lets {{ site.data.products.insomnia.name }} automatically refresh the token. When a token is close to expiration and cannot be refreshed automatically, {{ site.data.products.insomnia.name }} displays a warning message on the SCIM page and sends email notifications starting 20 days before the token expires.

Reviewer: ```suggestion
When a token is close to expiration and cannot be refreshed automatically, {{ site.data.products.insomnia.name }} displays a warning message on the SCIM page and sends email notifications starting 20 days before the token expires.

{% new_in 13 %} When SCIM is configured with OAuth 2.0, {{ site.data.products.insomnia.name }} automatically refresh the token. 
```
Just splitting these because that first sentence will apply to OAuth and other SCIM methods, but only the refresh applies to OAuth.

### Connector URL and token

When you enable SCIM in {{ site.data.products.insomnia.name }} from the Enterprise Controls, you generate a connector URL and token. This token authorizes your IdP to provision users and teams.

When generating the token, you can choose an expiration window. Available options include 30 days, 90 days, 180 days, 1 year, 2 years, or no expiration.

{:.warning}
> The token is displayed only once when it is generated. Store it securely. If you lose the token, refresh it in {{ site.data.products.insomnia.name }} and update the token in your IdP. To manually refresh the SCIM token, navigate to [**SCIM**](https://app.insomnia.rest/app/enterprise/scim).

### SCIM token management

When you enable SCIM, choose between these two options:

> - Automatic refresh (requires OAuth 2.0): SCIM tokens expire and {{ site.data.products.insomnia.name }} automatically refreshes them.

Author: ```suggestion
- Automatic refresh (requires OAuth 2.0): {{ site.data.products.insomnia.name }} automatically refreshes the SCIM tokens before they expire.
```

> - Automatic refresh (requires OAuth 2.0): {{ site.data.products.insomnia.name }} automatically refreshes the SCIM tokens before they expire.

Reviewer: ```suggestion
- Automatic refresh (requires OAuth 2.0) {% new_in 13 %}: {{ site.data.products.insomnia.name }} automatically refreshes the SCIM tokens before they expire.
```
